### PR TITLE
[C-536] Cache makeStyles StyleSheets

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Drastically reduces calls (up to 10x less) to `Stylesheet.create` by using style cache. This majorly improves renders for small items, like track-tiles that would otherwise each create their own stylesheet.

### CODE REVIEW

File: packages/mobile/package.json [hunk 1]
```diff
@@ -72,6 +72,7 @@
     "moment": "2.24.0",
     "node-libs-react-native": "1.2.1",
     "numeral": "2.0.6",
+    "object-hash": "^3.0.0",
```
Comment (raymondjacobson) [inline packages/mobile/package.json:75]: pin!

File: packages/mobile/src/styles/makeStyles.ts [hunk 2]
```diff
@@ -61,7 +64,14 @@ export const makeStyles = <PropsT, T extends NamedStyles<T> = NamedStyles<any>>(
     const stylesheet = useMemo(() => {
       const theme = { palette, typography, spacing, type: themeVariant }
       const namedStyles = styles(theme, memoizedProps)
-      return StyleSheet.create(namedStyles)
+      const namedStylesHash = hash(namedStyles)
```
Comment (raymondjacobson) [inline packages/mobile/src/styles/makeStyles.ts:67]: this is cool. 

is it worth us thinking about moving styles out of the render cycle? (not sure how we would do theme changes but maybe possible)

**Review by raymondjacobson — APPROVED** [review]

File: packages/mobile/package.json (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/package.json:75]: omg the number of times...

File: packages/mobile/src/styles/makeStyles.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/styles/makeStyles.ts:67]: yeah i would love to do that, but how do we handle theming? there doesn't seem to be a good pattern online for using Stylesheet directly like this, so might be worth considering emotionjs or some other css-in-jss-native option. long term question though imo.

Comment (raymondjacobson) [reply packages/mobile/src/styles/makeStyles.ts:67]: yes 100%

**Pull request merged**